src/ResourceManager.cpp

```diff
@@ -155,9 +160,8 @@ std::string ResourceManager::FindContainingArchiveFile(const std::string& filena
 {
 	for (const auto& archiveFile : ArchiveFiles)
 	{
-		int internalFileIndex = archiveFile->GetInternalItemIndex(filename);
-
-		if (internalFileIndex != -1) {
+		if (archiveFile->ContainsItem(filename)) {
+			std::size_t index = archiveFile->GetInternalItemIndex(filename);
```

Comment:
Small side note, but this code will do the name lookup twice. Once to see if the file exists, and then another to return the actual index. For code like this, a sentinel return value is more efficient as the lookup can be done once, while the returned value is used twice (does it exist? where does it exist?). For that reason, I often code both a sentinel return value version and an exception raising version of certain methods.

Though this is perfectly correct, and I think we should move on.

Reply:
Agree with this comment in general

Reply:
So I later found out `index` was unused and removed that line. That makes the side note largely irrelevant in this case.